pkg/queue/manager.go

```diff
@@ -296,7 +305,34 @@ func (m *Manager) deleteWorkloadFromQueueAndClusterQueue(w *kueue.QueuedWorkload
 	cq := m.clusterQueues[q.ClusterQueue]
 	if cq != nil {
 		cq.Delete(w)
+		if w.Spec.Admission != nil && m.queueAllInadmissibleWorkloadsInCohort(cq) {
```

Comment:
`deleteWorkloadFromQueueAndClusterQueue` is invoked from `DeleteWorkload` above. `DeleteWorkload` isn't called when the workload was admitted (because it is not supposed to be in the queue): https://github.com/kubernetes-sigs/kueue/blob/7b655a7a61d8d642ac4ae9ecfa45c99ab3bd70ba/pkg/controller/core/queuedworkload_controller.go#L134

but perhaps we should remove that condition and just call DeleteWorkload in all cases.

Reply:
I call the function `DeleteWorkload` in `manager.go` even if the workload was admitted.

Reply:
See https://github.com/kubernetes-sigs/kueue/pull/135#discussion_r838791361; I think it is better not to do it here and have an explicit API that we can evolve later to handle finer grained moving of workloads.